cmd/jaeger/internal/extension/jaegerstorage/extension_test.go

```diff
@@ -483,6 +485,27 @@ func TestCassandraError(t *testing.T) {
 	require.ErrorContains(t, err, "Servers: non zero value required")
 }
 
+func TestClickHouse(t *testing.T) {
+	testServer := clickhousetest.NewServer(clickhousetest.FailureConfig{})
+	t.Cleanup(testServer.Close)
+	ext := makeStorageExtension(t, &Config{
+		TraceBackends: map[string]TraceBackend{
+			"foo": {
+				ClickHouse: &clickhouse.Configuration{
+					Protocol: "http",
+					Addresses: []string{
+						testServer.Listener.Addr().String(),
+					},
+				},
+			},
+		},
+	})
+	ctx := context.Background()
```

Comment:
I recently learned there's such a thing as `t.Context()`, but need to be careful if using it in t.Cleanup() closures as I am not sure if it will be cancelled by then (defer closures should be fine)